client/scripts/views/components/admin_panel.ts

```diff
@@ -0,0 +1,634 @@
+import m, { Vnode } from 'mithril';
+import $ from 'jquery';
+import { OffchainThread, OffchainTag, CommunityInfo, RolePermission, ChainInfo, ChainNetwork, RoleInfo } from 'models';
+import { Button, Classes, Dialog, Icon, Icons, Tag, TagInput, ListItem, Table, Input, List, TextArea, Switch, Tabs, TabItem, RadioGroup, Form, FormGroup, FormLabel } from 'construct-ui';
+import app from 'state';
+import { sortAdminsAndModsFirst } from 'views/pages/discussions/roles';
+import User from './widgets/user';
+import 'components/admin_panel.scss';
+
+const RoleRow: m.Component<{ roledata?, onRoleUpdate?: Function }> = {
+  view: (vnode) => {
+    return (vnode.attrs.roledata?.length > 0) &&
+      m('RoleData', [
+          vnode.attrs.roledata?.map((role) => {
+            const chainOrCommObj = (app.activeChainId())
```

Comment:
It might be good to start by switching on community? I can't recall correctly but I thought it was possible that an offchain community had a default chain as well. If active chain uses that default chain then we may actually be in an offchain community when this thinks its a chain. Let me know if I'm wrong.

Reply:
I didn't realize this. I'll flip the logic accordingly, though it wasn't throwing any issues in the communities/chains I visited to QA the AdminPanel.